forta-uniswap-event-py/src/agent.py

```diff
@@ -0,0 +1,235 @@
+import binascii
+import json
+import os
+
+from web3 import logs, Web3
+
+from forta_agent import Finding, FindingType, FindingSeverity
+
+UNISWAP_V2_ROUTER_ADDR = None
+ETHER_THRESHOLD = None
+EVEREST_ID = None
+CONTRACT_INST = None
+
+
+class AttrDict(dict):
+    """
+    Allows a dictionary to act as a class object and attributes can be retrieved with
+    the . "dot" notation or the "['']" square bracket notation. This is a very useful helper
+    class since web3 typically uses square bracket notation and forta-agent uses dot syntax
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__dict__ = self
+
+
+def load_config():
+    """
+    Load the configuration values from config/agent-settings.json
+    """
+    global UNISWAP_V2_ROUTER_ADDR
+    global ETHER_THRESHOLD
+    global EVEREST_ID
+
+    dirname = os.path.dirname(os.path.abspath(__file__))
+    config_file = os.path.join(dirname, 'config', 'agent-settings.json')
+
+    with open(config_file, 'r') as f:
+        data = json.loads(f.read())
+
+    UNISWAP_V2_ROUTER_ADDR = Web3.toChecksumAddress(data['uniswap_v2_router_addr'])
+    ETHER_THRESHOLD = data['ether_threshold']
+    EVEREST_ID = data['everest_id']
+
+
+def get_contract_abi():
+    """
+    Given the address of a smart contract, return the abi provided by etherscan as a string
+    """
+    # ABI was retrieved from:
+    #   - https://etherscan.io/address/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#code
+    with open("weth_abi.json", "r") as f:
+        abi = json.loads(f.read())
+
+    return abi
+
+
+def get_contract_instance():
+    """
+    Get an instance of the contract by using the ABI saved in 'router_abi.json`
+    This will allow us to encode and decode function parameters
+    """
+    global CONTRACT_INST
+    if CONTRACT_INST:
+        return CONTRACT_INST
+
+    abi = get_contract_abi()
+    w3 = Web3()
+    CONTRACT_INST = w3.eth.contract(abi=abi)
+    return CONTRACT_INST
+
+
+def create_alert(to_addr, from_addr, amount_wad):
+    """
+    Return an alert with a metadata field that contains
+        - to address
+        - from address
+        - amount in wad
```

Comment:
"amount in wad" reads like wad is a unit or denomination--I think "amount in wad format" is more clear.  Also is the term "wad" commonly known?  Wordier but more clear: "amount (fixed-point value with 18 digits of precision)".

Reply:
wad is a fairly common term, so I think ill go with your first suggestion